### PR TITLE
LPS-126044 Content performance panel for all assets with display page template

### PR DESCRIPTION
**Pull request by cgoncas**
**Motivation**

As for 7.3 we've developed what can be considered an MVP for content performance (metrics for display page templates of web content), now we are aiming to retrieve and display these metrics for all kinds of pages: DPT for all assets but also content and widget pages.

**Proposed Solution**

1. The Content Performance icon will be visible for all display pages.

**How to test it**

1. Connect to AC 
2. Create a document DPT
3. Associate it with a document
3. The Content Performance icon should be visible
 
 
_To forward this pull, we need first https://github.com/liferay-tango/liferay-portal/pull/956 to be merged_

### CODE REVIEW

**Comment by darquesdev** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by darquesdev** [comment]
ci:test:relevant

**Comment by cgoncas** [comment]
ci:stop

**Comment by cgoncas** [comment]
Hey @darquesdev !!

We are having many problems with the CI, so we will only run ci:test:relevant when the pulls are ready to be forwarded.

Thanks!!

**Comment by darquesdev** [comment]
Perfect ;)

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+
+			Assert.assertEquals(
+				MockObject.class.getName(), infoItemReference.getClassName());
+		}
+		finally {
+			_classNameLocalService.deleteClassName(className);
+		}
+	}
+
+	public class MockObject {
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java:139]: This class 'MockObject' may be 'static', right?

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		_infoDisplayRequestAttributesContributor.addAttributes(
+			mockHttpServletRequest);
+
+		Assert.assertNull(
+			mockHttpServletRequest.getAttribute(
+				AnalyticsReportsWebKeys.INFO_ITEM_REFERENCE));
+	}
+
+	@Test
+	public void testAddAttributesWithLayoutDisplayPageObjectProvider() {
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java:65]: ❤️

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java (hunk 1 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-test/src/testIntegration/java/com/liferay/analytics/reports/layout/display/page/internal/request/attributes/contributor/test/AnalyticsReportsLayoutDisplayPageInfoDisplayRequestAttributesContributorTest.java:139]: yeahh

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-api/src/main/java/com/liferay/analytics/reports/layout/display/page/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+package com.liferay.analytics.reports.layout.display.page.info.item;
+
+import com.liferay.analytics.reports.info.item.AnalyticsReportsInfoItem;
+import com.liferay.layout.display.page.LayoutDisplayPageObjectProvider;
+
+/**
+ * @author Cristina González
+ */
+public interface LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem
+	extends AnalyticsReportsInfoItem<LayoutDisplayPageObjectProvider> {
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-api/src/main/java/com/liferay/analytics/reports/layout/display/page/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java:24]: `LayoutDisplayPageObjectProvider`  should be parametrized, right?

Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-api/src/main/java/com/liferay/analytics/reports/layout/display/page/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.java:24]: We can't :(, the tracker won't recognize a wildcard parametrized class, and this object won't be registered.

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java [hunk 4]
```diff
@@ -0,0 +1,424 @@
...
+ * @author Cristina González
+ */
+@Component(
+	service = {
+		AnalyticsReportsInfoItem.class,
+		LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem.class
+	}
+)
+public class LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl
+	implements AnalyticsReportsInfoItem<LayoutDisplayPageObjectProvider>,
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:81]: Would it be enough just implementing `LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem`?

Comment (darquesdev) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:81]: Ahh, ok, thanks for the explanation ;)

Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:81]: The same with LayoutDisplayPageObjectProviderAnalyticsReportsInfoItem,  the tracker doesn't have superclasses into account.

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+				infoFieldValue -> (Date)infoFieldValue.getValue()
+			).orElseGet(
+				() -> new Date()
+			);
+		}
+		catch (PortalException portalException) {
+			_log.error(portalException, portalException);
+		}
+
+		return new Date();
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:251]: Is this on purpose? I've seen your question in #t-lima

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java [hunk 6]
```diff
@@ -0,0 +1,424 @@
...
+					InfoField infoField = infoFieldValue.getInfoField();
+
+					return Objects.equals(
+						infoField.getInfoFieldType(),
+						DateInfoFieldType.INSTANCE);
+				}
+			).map(
+				infoFieldValue -> (Date)infoFieldValue.getValue()
+			).orElseGet(
+				() -> new Date()
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java]: Here, you can be replaced with method reference

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+			return Optional.ofNullable(
+				infoItemFieldValuesProvider.getInfoItemFieldValue(
+					layoutDisplayPageObjectProvider.getDisplayObject(), "title")
+			).filter(
+				infoFieldValue -> {
+					InfoField infoField = infoFieldValue.getInfoField();
+
+					return Objects.equals(
+						infoField.getInfoFieldType(),
+						DateInfoFieldType.INSTANCE);
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:277]: Should "title" use a `DateInfoFieldType` type?

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java (hunk 6 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java]: For now it is, for now, today will be the publish date, but we will fix it on the lima side :)

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java (hunk 5 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:251]: For now it is, for now, today will be the publish date, but we will fix it on the lima side :)

File: modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java (hunk 7 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-layout-display-page-impl/src/main/java/com/liferay/analytics/reports/layout/display/page/internal/info/item/LayoutDisplayPageObjectProviderAnalyticsReportsInfoItemImpl.java:277]: I would say that NO

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java [hunk 8]
```diff
@@ -134,21 +140,133 @@ public void testGetContext() throws Exception {
...
+
+				_mvcResourceCommand.serveResource(
+					mockLiferayResourceRequest, mockLiferayResourceResponse);
+
+				ByteArrayOutputStream byteArrayOutputStream =
+					(ByteArrayOutputStream)
+						mockLiferayResourceResponse.getPortletOutputStream();
+
+				JSONObject jsonObject = JSONFactoryUtil.createJSONObject(
+					new String(byteArrayOutputStream.toByteArray()));
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java:179]: Better just using `byteArrayOutputStream.toString()`, right?

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java [hunk 9]
```diff
@@ -134,21 +140,133 @@ public void testGetContext() throws Exception {
...
+				Assert.assertEquals(zonedDateTime.toLocalDate(), localDate);
+
+				Assert.assertEquals(
+					title, contextJSONObject.getString("title"));
+
+				JSONArray jsonArray = contextJSONObject.getJSONArray(
+					"viewURLs");
+
+				Assert.assertEquals(
+					String.valueOf(jsonArray), jsonArray.length(), 1);
```
Comment (darquesdev) [inline modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java:209]: The expected value should go first, right?

**Review by darquesdev — APPROVED** [review]
LGTM! Just minor comments

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java (hunk 8 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java]: Sure!!

File: modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java (hunk 9 above)
Comment (cgoncas) [reply modules/dxp/apps/analytics/analytics-reports-test/src/testIntegration/java/com/liferay/analytics/reports/web/internal/portlet/action/test/GetDataMVCResourceCommandTest.java]: yeah!!!

**Comment by cgoncas** [comment]
ci:test:sf

**Comment by cgoncas** [comment]
Continue in https://github.com/liferay-tango/liferay-portal/pull/970

**Pull request closed without merging**